programs/castle-vault/src/instructions/init_vault.rs

```diff
@@ -27,17 +20,18 @@ pub struct VaultConfigArg {
 }
 
 #[derive(Accounts)]
-#[instruction(bumps: InitBumpSeeds)]
+#[instruction(authority_bump: u8)]
 pub struct Initialize<'info> {
     /// Vault state account
     #[account(zero)]
     pub vault: Box<Account<'info, Vault>>,
 
     /// Authority that the vault uses for lp token mints/burns and transfers to/from downstream assets
+    /// CHECK: safe
     #[account(
         mut,
         seeds = [vault.key().as_ref(), b"authority".as_ref()],
-        bump = bumps.authority,
+        bump = authority_bump,
```

Comment:
Why is this bump needed?

Reply:
this isn't an init, so we use the provided bump as a constraint and then store it in the vault struct. using it as a constraint might not be necessary but we need to store it anyway so no harm no foul imo